wgpu/src/backend/direct.rs

```diff
@@ -1854,7 +1854,7 @@ impl crate::Context for Context {
                         channel: map_pass_channel(Some(&at.ops)),
                     })
             })
-            .collect::<ArrayVec<_, { wgc::MAX_COLOR_ATTACHMENTS }>>();
```

Comment:
Why does this need to be a vec too? I'm not sure of the exact data flow, but we should be able to preserve the arrayvecs everywhere

Reply:
Is it acceptable to validate directly within `command_encoder_begin_render_pass`?